core/DataAccess/LogQueryBuilder.php

```diff
@@ -31,9 +43,11 @@ public function getSelectQueryString(SegmentExpression $segmentExpression, $sele
             $bind = array_merge($bind, $segmentSql['bind']);
         }
 
-        $joins = $this->generateJoinsString($from);
-        $joinWithSubSelect = $joins['joinWithSubSelect'];
-        $from = $joins['sql'];
+        $tables = new JoinTables($this->logTableProvider, $from);
```

Comment:
Splitted `generateJoinsString ` and other methods into other classes so they are testable